### DISCUSSION TITLE
Retained MQTT messages are not cleared when MQTT is disabled, should they?

### DISCUSSION THREAD

**Discussion by reef-actor** [alienatedsec/yi-hack-v5#285] [Q&A]
**Describe the bug**
When MQTT is enabled there are some retained messages created such as motion_image, sound_detection etc.
When MQTT is disabled these are left in the broker. Would it make sense (is it good practice?) for these to be cleared from the broker?

**To Reproduce**
Steps to reproduce the behaviour:
1. Enable MQTT
2. Allow some retained messages to be sent
3. Disable MQTT
4. Retained messages are retained

**Expected behaviour**
Retained messages are cleared.

**Set Up Details (please complete the following information):**
 - Firmware Version: 0.4.0
 - Device: Yi Outdoor
 - Model: 9CUS

**Comment by alienatedsec** [comment]
Hi @reef-actor I would not expect retained messages to be cleared when MQTT is disabled on the camera. That would be a design of the MQTT that the broker retains the last message.

https://www.hivemq.com/blog/mqtt-essentials-part-8-retained-messages/

What would be your concern with retaining and clearing?
